crev-common/src/lib.rs

```diff
@@ -48,6 +48,49 @@ pub fn base64_encode<T: ?Sized + AsRef<[u8]>>(input: &T) -> String {
     base64::encode_config(input, base64::URL_SAFE_NO_PAD)
 }
 
+/// Takes an identifier like "https://crates.io" and converts it to something safe for use in paths etc.
+/// 
+/// # Examples
+/// 
+/// ```
+/// # use std::path::Path;
+/// # use crev_common::sanitize_name;
+/// // Pass through when able
+/// assert_eq!(sanitize_name("lazy_static"), Path::new("lazy_static"));
+/// 
+/// // Hash reserved windows filenames (or any other 3 letter name)
+/// assert_eq!(sanitize_name("CON"), Path::new("CON-f8d86fcc7f21486b"));
+/// 
+/// // Hash on escaped chars to avoid collisions
+/// assert_eq!(sanitize_name("https://crates.io"), Path::new("https___crates_io-c931072c02cbd3b6"));
+/// ```
+pub fn sanitize_name(s: &str) -> PathBuf {
+    let mut buffer = String::new();
+    let mut hash = s.len() == 3;
```

Comment:
Maybe we should just always append the hash? It's just more consistent, and since all of these paths are going to be internal, the human aesthetic pleasure is not of any concern.

Reply:
And that makes me thing of another attack - we should trim the names to something reasonable. 64characters maybe?

Reply:
Done :+1: .  The length limit is only so useful - concatenating 4 of these together will certainly exceed MAX_PATH - but at least you'll have to try harder to accomplish that, and so far I'm only concatenating 3 of these together ;)